src/pudl/analysis/service_territory.py

```diff
@@ -93,88 +93,77 @@ def get_census2010_gdf(pudl_settings, layer):
     return census_gdf
 
 
+def get_all_utils(pudl_out):
+    """
+    Compile IDs and Names of all known EIA Utilities.
+
+    Grab all EIA utility names and IDs from both the EIA 861 Service Territory table and
+    the EIA 860 Utility entity table. This is a temporary function that's only needed
+    because we haven't integrated the EIA 861 information into the entity harvesting
+    process and PUDL database yet.
+
+    Args:
+        pudl_out (pudl.output.pudltabl.PudlTabl): The PUDL output object which should be
+            used to obtain PUDL data.
+
+    Returns:
+        pandas.DataFrame: Having 2 columns ``utility_id_eia`` and ``utility_name_eia``.
+
+    """
+    return (
+        pd.concat([
+            pudl_out.utils_eia860()[["utility_id_eia", "utility_name_eia"]],
+            pudl_out.service_territory_eia861()[["utility_id_eia", "utility_name_eia"]],
+        ]).dropna(subset=["utility_id_eia"]).drop_duplicates(subset=["utility_id_eia"])
+    )
+
+
 ################################################################################
 # Functions that compile geometries based on EIA 861 data tables:
 ################################################################################
-def get_balancing_authority_counties(ba_ids,
-                                     st_eia861,
-                                     ba_assn_eia861,
-                                     limit_by_state=True):
+def get_territory_fips(ids, assn, assn_col, st_eia861, limit_by_state=True):
     """
-    Compile counties associated with select balancing authorities by year.
+    Compile county FIPS codes associated with an entity's service territory.
 
-    For each balancing authority identified by ba_ids, look up the set of counties
-    associated with that BA on an annual basis. Optionally limit the set of counties
-    to those within states where the selected balancing authorities have been seen
-    in association with each utility whose counties make up the BA territory.
+    For each entity identified by ids, look up the set of counties associated
+    with that entity on an annual basis. Optionally limit the set of counties
+    to those within states where the selected entities reported activity elsewhere
+    within the EIA 861 data.
 
     Args:
-        ba_ids (iterable of ints): A collection of EIA balancing authority IDs.
+        ids (iterable of ints): A collection of EIA balancing authority IDs.
```

Comment:
I am still confused as to why you'd want to restrict these ids... is it so that in the `ferc714.Respondents` object you can choose to prioritize utils over BA's? why not handle that over there and not require this as an input? Excuse this comment if you've already explained it or it isn't pertinent.

Reply:
I'm not sure what you mean by "restrict these IDs." Do you mean why not always get the FIPS codes for every entity and drop the unwanted ones somewhere else?  Right now we're just doing the simple case of taking all the BA and all the Util IDs, and assuming that one of them should take precedence. But if that turns out to be wrong (which it almost certainly is), being able to limit the set of Util or BA IDs that are considered will let us fine tune the categorization. Like if we know that Xcel Colorado should be treated as a Utility, not a BA, even though its ID shows up in both sets of IDs, and we're prioritizing Balancing Authorities, then we don't include its ID in the list of BA IDs.

Reply:
I understand that you'll want to be able to treat/categorize specific respondents differently, but I guess it feels odd to me to need to pass those ids around in these functions instead of dealing with that categorization in one place (presumable in the ferc714.Respondents object). this is not a blocking concern for me on this PR, just something that keeps nagging at me.

Reply:
I was also thinking it'd be nice to be able to just get the FIPS ids (or geometries) for a single entity at a time, if you're not trying to do the *entire* dataset.  And in the case of FERC 714, a large majority of all the entities that exist in EIA just don't show up at all, so it seemed more efficient to only grab the data for the entities that are actually being used, rather than everything, every time.

Reply:
Hmm, okay. That makes sense but I guess it seems like it would make this harder to interact with. But you certainly interact with this more than I do. Perhaps if it were me the default setting would be to do all the the ids, so you could if you wanted to curate the ids to do one at a time or do only the ones you have assigned to be utility. But you wouldn't have to prep the ids every time.

Reply:
On the ferc714 Respondents class side of things, where this is primarily being used right now, I do have it set to, by default, use the entire universe of BA and Utility IDs.